tests/functional/test_hydro_heuristic_complex_case.py

```diff
@@ -0,0 +1,542 @@
+
+    capacity = 1711510
+
+    for scenario in range(scenarios):
+        initial_level = 0.5 * capacity
+
+        # Répartition des apports mensuels
+        (
+            monthly_demand,
+            monthly_inflow,
+            monthly_max_generating,
+        ) = get_all_data(scenario, "monthly")
+        monthly_target = get_target(monthly_demand, sum(monthly_inflow))
+
+        # Ajustement de la réapartition mensuelle
+        problem = create_hydro_problem(
+            horizon="monthly",
+            target=monthly_target,
+            inflow=monthly_inflow,
+            max_generating=monthly_max_generating,
+            initial_level=initial_level,
+        )
+
+        status, monthly_generation, _ = solve_hydro_problem(problem)
+
+        assert status == problem.solver.OPTIMAL
+
+        all_daily_generation: List[float] = []
+        day_in_year = 0
+        (
+            daily_demand,
+            daily_inflow,
+            daily_max_generating,
+        ) = get_all_data(scenario, "daily")
+
+        for month in range(12):
+            number_day_month = get_number_of_days_in_month(month)
+            # Répartition des crédits de turbinage jour par jour
+
+            daily_target = get_target(
+                demand=daily_demand[day_in_year : day_in_year + number_day_month],
+                total_target=monthly_generation[month],
+                inter_breakdown=interdaily_breakdown,
+            )
+            # Ajustement de la répartition jour par jour
+            problem = create_hydro_problem(
+                horizon="daily",
+                target=daily_target,
+                inflow=daily_inflow[day_in_year : day_in_year + number_day_month],
+                max_generating=daily_max_generating[
+                    day_in_year : day_in_year + number_day_month
+                ],
+                initial_level=initial_level,
+            )
+
+            status, daily_generation, initial_level = solve_hydro_problem(problem)
+
+            assert status == problem.solver.OPTIMAL
+
+            all_daily_generation = all_daily_generation + daily_generation
+            day_in_year += number_day_month
+
+        # Calcul des cibles hebdomadaires
+        weekly_target = calculate_weekly_target(
+            all_daily_generation,
+        )
+
+        # Vérification des valeurs trouvées
+        expected_output_file = open(
+            "tests/functional/data_complex_case/hydro/values-weekly.txt",
+            "r",
+        )
+        expected_output = expected_output_file.readlines()
+        for week in range(52):
+            assert float(expected_output[week + 7].strip().split("\t")[9]) == round(
+                weekly_target[week]
+            )
+
+
+def calculate_weekly_target(all_daily_generation: list[float]) -> list[float]:
+    weekly_target = np.zeros(52)
+    week = 0
+    day_in_week = 0
+    day_in_year = 0
+
+    while week < 52:
+        weekly_target[week] += all_daily_generation[day_in_year]
```

Comment:
Ça marcherai qqch comme ça ? 

`weekly_target[week] = sum(all_daily_generation[day] for day in range(7 * week , 7 * (week + 1))`

Plus besoin de faire les incrémentations à la main et tu peux remplacer le `while` par `for week in range(weekly_target.size())`